qtranslate_core.php

```diff
@@ -1143,6 +1129,9 @@ function qtranxf_convertURL( $url = '', $lang = '', $forceadmin = false, $showDe
         $showDefaultLanguage = ! $q_config['hide_default_language'];
     }
     $showLanguage = $showDefaultLanguage || $lang != $q_config['default_language'];
+
+    $url = apply_filters( 'qtranslate_convert_url_before', $url, $lang );
+
```

Comment:
`$showLanguage` is ignored here. I don't understand the logic. If a slug is available, why do we even call the next function `qtranxf_get_url_for_language` ?

Reply:
`qtranxf_get_url_for_language `is the general function to provide the url for that language, passed url being filtered before or not.
In case of slugs module is active (which however does not always mean filter is there, as `qtranxf_convertURL` is also called before slugs filter is added) , it seems it is still needed to handle `showDefaultLanguage` setting.